File: iOS/PengChauFerryTests/PreferencesTest.swift
//
//  PreferencesTest.swift
//  PengChauFerryTests
//
//  Created by Lionello Lunesu on 11/4/2020.
//  Copyright © 2020 Lionello Lunesu. All rights reserved.
//

import XCTest

@testable import PengChauFerry

class PreferencesTest: XCTestCase {

    func testLanguage() {
        let lang = "en"
        Preferences().language = lang
        XCTAssertEqual(lang, Preferences().language)

        Preferences().language = nil
        XCTAssertNil(Preferences().language)
    }

    func testLastRefresh() {
        let now = LocalDateTime(timeIntervalSince1970: 42.0)
        Preferences().lastRefresh = now
        XCTAssertEqual(now, Preferences().lastRefresh)

        Preferences().lastRefresh = nil
        XCTAssertNil(Preferences().lastRefresh)
    }

}

File: Android/app/src/androidTest/java/com/lunesu/pengchauferry/PreferencesTest.kt
package com.lunesu.pengchauferry

import androidx.test.ext.junit.runners.AndroidJUnit4
import androidx.test.platform.app.InstrumentationRegistry
import org.joda.time.LocalDateTime
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNull
import org.junit.Test
import org.junit.runner.RunWith

/**
 * Instrumented test, which will execute on an Android device.
 *
 * See [testing documentation](http://d.android.com/tools/testing).
 */
@RunWith(AndroidJUnit4::class)
class PreferencesTest {
    private val appContext = InstrumentationRegistry.getInstrumentation().targetContext

    @Test
    fun useAppContext() {
        // Context of the app under test.
        assertEquals("com.lunesu.pengchauferry", appContext.packageName)
    }

    @Test
    fun testLanguage() {
        val lang = "en"
        Preferences(appContext).language = lang
        assertEquals(lang, Preferences(appContext).language)

        Preferences(appContext).language = null
        assertNull(Preferences(appContext).language)
    }

    @Test
    fun testLastRefresh() {
        val now = LocalDateTime(42)
        Preferences(appContext).lastRefresh = now
        assertEquals(now, Preferences(appContext).lastRefresh)

        Preferences(appContext).lastRefresh = null
        assertNull(Preferences(appContext).lastRefresh)
    }

    /*@Test
    fun testClear() {
        val db = DbOpenHelper(null)
        val vm = FerryViewModel(Application(), db)
        val viewModelStore = ViewModelStore()
        val viewModelProvider = ViewModelProvider(viewModelStore, object : ViewModelProvider.Factory {
            @Suppress("UNCHECKED_CAST")
            override fun <T : ViewModel?> create(modelClass: Class<T>): T = vm as T
        })
        viewModelProvider.get(vm::class.java)

        //Run 2
        viewModelStore.clear()//To call clear() in ViewModel

        Assert.assertFalse(db.readableDatabase.isOpen)
        Assert.assertFalse(db.writableDatabase.isOpen)
    }*/
}
